### PR TITLE
Adds the Detective Holobarrier Projector + Chalk

### PR DESCRIPTION
**Pull request by ItsMarmite**
image

## What Does This PR Do
We've got this nifty little update, see? We're tossing in the detective holobarrier projector into the detective's stash, swapping out the old security gizmo. Now, this projector? It's got some perks. Deploys quicker, handles more active barriers simultaneously. And get this, these barriers come with a special look – real unique sprite. Oh, and here's the kicker – they give a little jolt if someone tries to mess with 'em.

But that's not all. We're throwing in a new crayon type too, a real fancy one. It's called chalk, and it's taking the place of the old white crayon in the detective's toolkit. This bad boy draws faster. So, just a heads up on these changes. You know, just making sure everything's top-notch for our detectives out there.

## Why It's Good For The Game
But, just one more thing. You know, I've been working this detective gig for a while now, and let me tell ya, setting up a crime scene can be a real headache. Things get hectic, rounds start speeding up, and don't even get me started on assistants barging through my barriers like they're on a mission. It's a mess, really. You spend all this time laying out your crime scene, only for someone to come along and mess it up. Now, you gotta start from scratch.

Now, here's the deal. I'm thinking this shock feature on the barriers might just solve that problem. You see, if folks know there's a little zap waiting for 'em, maybe they'll think twice about crashing through crime scenes. It's all about giving the detective a chance to shine, you know? I just want to do some good ol' detective roleplay, draw those classic white chalk outlines, and spill my thoughts out loud.

But, and here's the twist, it's a double-edged sword. Those sneaky antags, they could swipe the projector and set up their own shocking barriers. Imagine the chaos for the security team if they fall for that trap. So, it's a bit of a gamble, but hey, that's the game we're in. Gotta keep everyone on their toes, I suppose.
## Images of changes
image

## Testing
Shocked myself, deployed barriers, drew shit all over the floor. Checked to see if the barriers did not bypass insuls.

## Changelog
:cl:
add: Adds the Detective Holosign Projector.
add: Detectives now spawn with a piece of chalk.
/:cl:

### CODE REVIEW

**Comment by Ionward** [comment]
i hope i speak for everyone when i say i appreciate the columbo commitment

**Comment by SynthTwo** [comment]
based columbo ref, speedmerge this

File: code/game/objects/items/crayons.dm [hunk 1]
```diff
@@ -240,6 +240,11 @@
 	colour = "#FFFFFF"
 	colourName = "white"
 
+/obj/item/toy/crayon/white/chalk
+	name = "detective's chalk"
+	desc = "A stick of white chalk for marking crime scenes."
```
Comment (lewcc) [inline code/game/objects/items/crayons.dm:245]: Should probably make it plural so it'll be displayed like `that's some detective's chalk.`'
```suggestion
	desc = "A stick of white chalk for marking crime scenes."
	gender = PLURAL
```

File: code/game/objects/items/weapons/holosign_projector.dm [hunk 2]
```diff
@@ -93,6 +93,16 @@
 	holosign_type = /obj/structure/holosign/barrier
 	creation_time = 30
 
+/obj/item/holosign_creator/detective
+	name = "detective holobarrier projector"
+	desc = "A holographic projector that creates shocked investigation barriers."
+	icon_state = "signmaker_det"
+	belt_icon = null
+	holosign_type = /obj/structure/holosign/barrier/cyborg/hacked/detective
+	creation_time = 10
```
Comment (lewcc) [inline code/game/objects/items/weapons/holosign_projector.dm:102]:
```suggestion
	creation_time = 1 SECONDS
```

**Comment by Henri215** [comment]
It shocks if you bump into the barrier? Or only when you touch it? Its supposed to be used during active combat too?

**Comment by ItsMarmite** [comment]
> It shocks if you bump into the barrier? Or only when you touch it? Its supposed to be used during active combat too?

Shocks when you move through it, or touch it. Can be taken down with the projector. While it could be used in active combat you have to stand still and can only make one at a time, and security aren't shockproof unlike engineers. It does roughly 7 burn and knocks down per 'shock'

**Comment by Gatchapod** [comment]
Let's chalk it up to the shocking developments in the forensic science.

**Review by 1080pCat — APPROVED** [review]
The code looks fine.
However i have some worries that the shocking will mainly be used to screw over antags in traps. also shove spamming takes ~18 seconds to kill while also stunning for a moment..
dreamseeker_sQtiSLMPvz
As its a unique item its *probably* fine though.. probably..

**Comment by ItsMarmite** [comment]
> The code looks fine. However i have some worries that the shocking will mainly be used to screw over antags in traps. also shove spamming takes ~18 seconds to kill while also stunning for a moment.. dreamseeker_sQtiSLMPvz [ dreamseeker_sQtiSLMPvz ](https://private-user-images.githubusercontent.com/96908085/311503075-e4131026-7e74-43a1-85e3-a389c250ddc1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAwNjc1NjksIm5iZiI6MTcxMDA2NzI2OSwicGF0aCI6Ii85NjkwODA4NS8zMTE1MDMwNzUtZTQxMzEwMjYtN2U3NC00M2ExLTg1ZTMtYTM4OWMyNTBkZGMxLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEwVDEwNDEwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0ZjllMGNhOWQ1MjFlNzg1OTRhYmE0N2I3MWVhMDRhZWY1ZTEzNjNlYmQwZmRiMzE4ZWY4M2U0NmYyMDA1YzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.a5WZvZykDhYwe7Si0XtWUl9JHT2-X8eEgIWzQhptmwA) [ ](https://private-user-images.githubusercontent.com/96908085/311503075-e4131026-7e74-43a1-85e3-a389c250ddc1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAwNjc1NjksIm5iZiI6MTcxMDA2NzI2OSwicGF0aCI6Ii85NjkwODA4NS8zMTE1MDMwNzUtZTQxMzEwMjYtN2U3NC00M2ExLTg1ZTMtYTM4OWMyNTBkZGMxLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEwVDEwNDEwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0ZjllMGNhOWQ1MjFlNzg1OTRhYmE0N2I3MWVhMDRhZWY1ZTEzNjNlYmQwZmRiMzE4ZWY4M2U0NmYyMDA1YzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.a5WZvZykDhYwe7Si0XtWUl9JHT2-X8eEgIWzQhptmwA) As its a unique item its _probably_ fine though.. probably..

See my conversation with you in the Discord. I think there's a lot of ways the Detective could kill someone and htis one isn't really all that efficient.

**Review by lewcc — APPROVED** [review]

**Comment by Kenionatus** [comment]
Is it intentional that the projector cannot be put into the sec belt?

**Pull request merged**